src/services/OswServiceTaskWiFi.cpp

```diff
@@ -42,23 +42,53 @@ void OswServiceTaskWiFi::loop() {
+            // Nice - reset timeout
+            this->m_connectTimeout = 0;
+            this->m_connectFailureCount = 0;
+        } else if(!this->m_connectTimeout and WiFi.status() != WL_CONNECTED) {
+            // Wifi is either disconnected or unavailable...
+            if(this->onlyOneModeSimultaneously and this->m_enableStation) {
+                // This watch does not support connecting to networks with an active AutoAP.
+                // It would therefore be not helpful to activate the timeout to try an other
+                // action, while the AutoAP is active as this will shutdown the AutoAP as the
+                // timeout expires!
+            } else {
+                // -> start timeout
+                this->m_connectTimeout = time(nullptr);
 #ifndef NDEBUG
-            Serial.println(String(__FILE__) + ": [AutoAP] Timeout activated: 10 seconds");
+                Serial.println(String(__FILE__) + ": [Connection] Timeout activated: 10 seconds");
 #endif
+            }
         }
 
-        if(OswConfigAllKeys::wifiAutoAP.get() && !this->m_enableStation and this->m_autoAPTimeout and this->m_autoAPTimeout < time(nullptr) - 10) { //10 seconds no network -> auto ap!
-            this->enableStation();
-            this->m_enabledStationByAutoAP = time(nullptr);
+        // Handling in case of 10 seconds without a successful connect
+        if(this->m_connectTimeout and this->m_connectTimeout < time(nullptr) - 10) {
+            ++this->m_connectFailureCount;
 #ifndef NDEBUG
-            Serial.println(String(__FILE__) + ": [AutoAP] Active for " + String(this->m_enabledStationByAutoAPTimeout) + " seconds (password is " + this->m_stationPass + ").");
+            Serial.println(String(__FILE__) + ": [Connection] Timeout expired. Looking for alternatives (" + this->m_connectFailureCount + ")...");
 #endif
+            this->m_connectTimeout = 0; // Disable connection timeout, we will re-enable it with the next iteration, if we still want to connect (and are not in AutoAP mode)
+
+            // We have four options if our connection times out; the fourth is the auto-ap, which is may not available
+            if(this->m_connectFailureCount % 4 == 3) {
+                if(OswConfigAllKeys::wifiAutoAP.get()) {
+                    if(!this->m_enableStation) {
+                        this->enableStation();
+                        this->m_enabledStationByAutoAP = time(nullptr);
+#ifndef NDEBUG
+                        Serial.println(String(__FILE__) + ": [AutoAP] Active for " + String(this->m_enabledStationByAutoAPTimeout) + " seconds (password is " + this->m_stationPass + ").");
+#endif
+                    }
+                } else {
+#ifndef NDEBUG
+                    Serial.println(String(__FILE__) + ": [AutoAP] Is disabled; no action taken.");
+#endif
+                }
+            } else {
+                this->selectCredentials();
+                this->connectWiFi();
```

Comment:
It is already in the function to review the credentials.

Reply:
Whoops! Fixed.